api/repositories/service_plan_repository_test.go

```diff
@@ -600,5 +602,96 @@ var _ = Describe("ServicePlanRepo", func() {
 				})
 			})
 		})
+
+		Describe("DeletePlanVisibility", func() {
+			BeforeEach(func() {
+				cfServicePlan := &korifiv1alpha1.CFServicePlan{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: rootNamespace,
+						Name:      planGUID,
+					},
+				}
+				Expect(k8s.PatchResource(ctx, k8sClient, cfServicePlan, func() {
+					cfServicePlan.Spec.Visibility.Type = visibilityType
+					cfServicePlan.Spec.Visibility.Organizations = []string{cfOrg.Name, anotherOrg.Name}
+				})).To(Succeed())
+			})
+
+			When("the user is not authorized", func() {
+				BeforeEach(func() {
+					visibilityErr = repo.DeletePlanVisibility(ctx, authInfo, repositories.DeleteServicePlanVisibilityMessage{
```

Comment:
This could go to a Describe-global `JustBeforeEach` as it is invoking the code under test. This will also reduce duplication